lib/jxl/enc_adaptive_quantization.cc

```diff
@@ -877,15 +883,15 @@ void FindBestQuantization(const ImageBundle& linear, const Image3F& opsin,
     iters = 2;
   }
   for (int i = 0; i < iters + 1; ++i) {
-    if (FLAGS_dump_quant_state) {
-      printf("\nQuantization field:\n");
-      for (size_t y = 0; y < quant_field.ysize(); ++y) {
-        for (size_t x = 0; x < quant_field.xsize(); ++x) {
-          printf(" %.5f", quant_field.Row(y)[x]);
-        }
-        printf("\n");
+#if JXL_DEBUG_ADAPTIVE_QUANTIZATION
```

Comment:
Let's avoid `#if` to skip compiling code and use `if (some_constexpr_boolean)` instead.

Reply:
ok — there are probably quite a few other spots where we were already doing that, feel free to update those too :)